jenkins/scripts/fetch-failed-request-logs.py

```diff
@@ -0,0 +1,200 @@
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "--date",
+        dest="date",
+        help="Date to examine for failed Cachito requests, in Year-Mon-Date format.  "
+        + "Example: 2021-11-16",
+    )
+    group.add_argument(
+        "--yesterday",
+        action="store_true",
+        help="Return failures from yesterday.  Causes script to ignore the --date argument",
+    )
+    parser.add_argument("--server", dest="cachitoserver", help="Cachito Server Hostname",
+                        required=True)
+    return parser.parse_args()
+
+
+def _is_same_date(item, request_date):
+    date_obj = datetime.datetime.strptime(item["updated"], "%Y-%m-%dT%H:%M:%S.%f").date()
+    if request_date == date_obj:
+        return 1
+
+
+def _fetch_cachito_jobs(cachitoserver, request_date, state):
+    # Cachito response for requests is reverse chronological order, so newest fist.
+
+    # API args
+    #
+    # -  page=int
+    # -  per_page=int
+    # -  state=string
+    # -  created_from=datetime
+    # -  created_to=datetime
+    #
+
+    # Cachito config currently has hard limit of 100.
+    per_page_requests = 100
+    page = 1
+    http = urllib3.PoolManager()
+    cachito_failed_requests = []
+    cachito_total_requests = []
+    api_url = "http://" + cachitoserver + "/" + API_PATH
+
+    # Initial URL.  The meta section of the response gives the 'next' URL with
+    # an updated page value, and 'None' when at the end of the available
+    # requests based on the query.  Using the 'next' URL to simplify the logic here.
+    request_url = (
+        api_url
+        + "/requests"
+        + "?per_page="
+        + str(per_page_requests)
+        + "&page="
+        + str(page)
+        + "&created_from="
+        + str(request_date)
+        + "&created_to="
+        + str(request_date)
+    )
+    while True:
+        r = http.request("GET", ic(request_url))
+        json_response = json.loads(r.data)
+        ic(json_response)
+
+        cachito_total_requests.extend(
+            [item for item in json_response["items"] if _is_same_date(item, request_date)]
+        )
+        cachito_failed_requests.extend(
+            [
+                item
+                for item in json_response["items"]
+                if _is_same_date(item, request_date) and item["state"] == state
+            ]
+        )
+        if json_response["meta"]["next"] is None:
+            ic("Received None for next URL... breaking")
+            break
+        request_url = (
+            json_response["meta"]["next"]
+            + "&created_from="
+            + str(request_date)
+            + "&created_to="
+            + str(request_date)
+        )
+
+    ic(cachito_total_requests)
+    ic(cachito_failed_requests)
+
+    return (
+        cachito_failed_requests,
+        len(cachito_failed_requests),
+        len(cachito_total_requests),
+    )
+
+
+def _fetch_cachito_logs(cachitoserver, id):
```

Comment:
If you remove `id` from the definition it would still work.
```
def _fetch_cachito_logs(cachitoserver):
    ic("Fetching logs for id " + str(id))
```
```
Fetching logs for id <built-in function id>
```

Reply:
Didn't work for me...  I'm going to leave this one alone.

ic| "Fetching logs for id " + str(id): 'Fetching logs for id <built-in function id>'
ic| str(r.data.decode("utf-8")): '{"error":"The requested resource was not found"}